src/modules/ctc/src/ctc.cc

```diff
@@ -184,6 +231,36 @@ types::Error Ctc::GetTrainPointerById(const types::TrainId train_id, std::shared
     return error;
 }
 
+types::Error Ctc::GetTrainPointerFromBlockOccupancy(const types::BlockId block_id, std::shared_ptr<ctc::Train> &train_pointer)
```

Comment:
These methods that get shared pointers to trains and blocks are unnecessary since everything in the CTC class has access to these lists of trains and blocks.  The same thing could easily be accomplished with `find_if` which would be a lot cleaner, simpler, and less error-prone in my opinion.  Also, be careful creating shared pointers to objects that have already been allocated.  The whole point of a shared pointer is to manage the lifetime of an object, but in this case, the lifetime of the objects the shared pointers point to is managed by the vectors they are contained in, not the shared pointer itself.

Reply:
removed. find_if is indeed easier without having extra getter methods